sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AzureKeyCredentialPolicy.java

```diff
@@ -47,23 +38,64 @@ protected void beforeSendingRequest(HttpPipelineCallContext context) {
-        Objects.requireNonNull(credential, "'credential' cannot be null.");
+        this(name, credential, null);
+    }
+
+    /**
+     * Creates a policy that uses the passed {@link AzureKeyCredential} to set the specified header name.
+     * <p>
+     * The {@code prefix} will be applied before the {@link AzureKeyCredential#getKey()} when setting the header. A
+     * space will be inserted between {@code prefix} and credential.
+     *
+     * @param name The name of the key header that will be set to {@link AzureKeyCredential#getKey()}.
+     * @param credential The {@link AzureKeyCredential} containing the authorization key to use.
+     * @param prefix The prefix to apply before the credential, for example "SharedAccessKey credential".
+     * @throws NullPointerException If {@code name} or {@code credential} is {@code null}.
+     * @throws IllegalArgumentException If {@code name} is empty.
+     */
+    public AzureKeyCredentialPolicy(String name, AzureKeyCredential credential, String prefix) {
+        this(validateName(name), Objects.requireNonNull(credential, "'credential' cannot be null."), prefix);
+    }
+
+    private static HttpHeaderName validateName(String name) {
         Objects.requireNonNull(name, "'name' cannot be null.");
         if (name.isEmpty()) {
             throw LOGGER.logExceptionAsError(new IllegalArgumentException("'name' cannot be empty."));
         }
 
-        this.name = HttpHeaderName.fromString(name);
+        return HttpHeaderName.fromString(name);
+    }
+
+    AzureKeyCredentialPolicy(HttpHeaderName name, AzureKeyCredential credential, String prefix) {
+        this.name = name;
         this.credential = credential;
+        this.prefix = prefix;
     }
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        return inner.process(context, next);
+        if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
+            return FluxUtil.monoError(LOGGER,
+                new IllegalStateException("Key credentials require HTTPS to prevent leaking the key."));
+        }
+
+        setCredential(context.getHttpRequest().getHeaders());
+        return next.process();
     }
 
     @Override
     public HttpResponse processSync(HttpPipelineCallContext context, HttpPipelineNextSyncPolicy next) {
-        return inner.processSync(context, next);
+        if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalStateException("Key credentials require HTTPS to prevent leaking the key."));
+        }
+
+        setCredential(context.getHttpRequest().getHeaders());
+        return next.processSync();
+    }
+
+    void setCredential(HttpHeaders headers) {
+        String credential = this.credential.getKey();
+        headers.set(name, (prefix == null) ? credential : prefix + " " + credential);
```

Comment:
Remove trailing space in case the prefix already had one.
```suggestion
        headers.set(name, (prefix == null) ? credential : prefix.trim() + " " + credential);
```